Documents/AnalyticsToolDesignDocument.md

```diff
@@ -0,0 +1,48 @@
+# VFS For Git: Analytics Tooling Design Document
+
+This document details the design of and plan for the **VFS4G analytics tool**. The tool aims to help users identify reasons why their enlistment may be slowing down and the `gvfs` commands running slowly. By collecting and reporting statistics, the power to uncover and hopefully fix problems as they arrive moves to the user, empowering them to use the software to its fullest potential.
+
+## Problem
+
+When users unintentionally hydrate too many files within their enlistment, commands like checkout and commit increase their running time by several orders of magnitude. This causes unacceptable losses in performance and greatly hinders users as they go about their daily development cycle. The primary metric of the overhydration is the number of placeholder files which have been pulled from the remote into the local system. Slowdowns may also stem from having many modified files or directories which increases the amount of work that git does under the hood. Additionally, if a local enlistment has been in use for a long time, the number of loose object may eventually reach the order of several thousand in which case that also contributes to the loss in performance.
+
+## Provided Statistics
+
+### Safe to access
+
+- **Modified paths** - The number of files which have been edited and have been handed over to git
+- **Git index** - The differences between modified paths and the index tracked by git

```

Comment:
Not sure what this means or what you will be using the index for.

Reply:
I added the following to the end of the git index bullet point, it might clear things up.

"This might be useful in determining what differences the local repository has to predict the complexity and potentially slowdowns of an upcoming commit."

It is also possible however that this is information which could easily be accessed elsewhere or maybe even that everything within the modified paths will always align perfectly with the index (which is suspect might be the case)